sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1243,6 +1244,12 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
         IsNotNull(e)
       case SqlBaseParser.NULL =>
         IsNull(e)
+      case SqlBaseParser.TRUE =>
+        invertIfNotDefined(BooleanTest(e, Some(true)))
+      case SqlBaseParser.FALSE =>
+        invertIfNotDefined(BooleanTest(e, Some(false)))
+      case SqlBaseParser.UNKNOWN =>
+        invertIfNotDefined(BooleanTest(e, None))
```

Comment:
I'm still reluctant for this as I mentioned before in my previous comments. The current PR direction is not aligned to the advice.

Reply:
I don't known the reason that must use `IsNull` or `IsNotNull`. I think boolean predicate and null predicate is two different syntax.